server/instance_cloud_oauth.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	jira "github.com/andygrunwald/go-jira"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/mattermost/mattermost-plugin-jira/server/utils"
+	"github.com/mattermost/mattermost-plugin-jira/server/utils/types"
+)
+
+type cloudOAuthInstance struct {
+	*InstanceCommon
+
+	// The SiteURL may change as we go, so we store the PluginKey when as it was installed
+	MattermostKey string
+
+	JiraResourceID   string
+	JiraClientID     string
+	JiraClientSecret string
+}
+
+type CloudOAuthConfigure struct {
+	InstanceURL  string `json:"instance_url"`
+	ClientID     string `json:"client_id"`
+	ClientSecret string `json:"client_secret"`
+}
+
+type JiraAccessibleResources []struct {
+	ID string
+}
+
+var _ Instance = (*cloudOAuthInstance)(nil)
+
+const (
+	JiraScopes       = "read:jira-user,read:jira-work,write:jira-work"
```

Comment:
What exactly is `jira-work`, and does this allow the user to create issues and comments? We'll also want to be able to assert that the user has "browse project" permission for a given project, to assert the user has permissions to read issues from a given project.

Reply:
Yes `jira-work` permission allows users to view/create issues and comments. 

<img width="1012" alt="Screenshot 2023-04-04 at 3 36 19 PM" src="https://user-images.githubusercontent.com/1429138/229759235-e1bc1ced-33e9-4bd8-84e6-b57aa3ac882a.png">

There's no specific permission for `browse project`. The closest would be `manage project`, but that does not seems appropriate as per what we want to achieve.

Reply:
Yes it seems `read:jira-work` satisfies the issue based on your screenshot :+1: